snuba/consumers/consumer_builder.py

```diff
@@ -93,12 +96,20 @@ def __init__(
             .topic
         )
 
+        if slice_id is not None:
+            # there should be some sort of assertion to check
+            # whether the slice_id passed is valid for this storage
+            physical_topic = SLICED_KAFKA_TOPIC_MAP[(topic.value, slice_id)]
+            # this will require registration of the physical topic
+            topic = register_topic(physical_topic)
```

Comment:
I wonder if `register_topic` itself could validate whether the Kafka topic configuration exists and is present on the configured broker

Reply:
I'm not sure if I can modify `register_topic` for this, but I think I can add the validation in the consumer builder itself as soon as it has the storage and slice_id.